hr_employee_seniority/models/hr_employee.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+# copyright 2013 Michael Telahun Makonnen <mmakonnen@gmail.com>
+# copyright 2017 Denis Leemann, Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import datetime
+
+from odoo import api, fields, models
+from odoo import tools
+
+
+class HrEmployee(models.Model):
+
+    _inherit = 'hr.employee'
+
+    initial_employment_date = fields.Date(
+        string='Initial Date of Employment',
+        help='Date of first employment if it was before the start of the '
+             'first contract in the system.',
+    )
+    length_of_service = fields.Float(
+        'Lenght of Service',
+        compute='_compute_months_service',
+    )
+
+    @api.depends('contract_ids')
```

Comment:
`initial_employment_date` to add in dependencies (a date could be corrected thus the cache should be updated with a correct value in `length_of_service` which then could be directly displayed on UI without having to refresh the page)